pkgs/tools/system/hddfancontrol/default.nix

```
@@ -0,0 +1,45 @@
+{ lib, fetchpatch, python3Packages, fetchFromGitHub, hddtemp, hdparm, smartmontools }:
+
+let pythonPackages = python3Packages; in
+
+with { inherit (pythonPackages) buildPythonPackage; };
+
+buildPythonPackage rec {
```

Although I think you are free to do it this way (i.e. there are no rules saying you're not allowed to), it creates a lot of unnecessary abstraction. There are no cases of `pythonPackages` being anything else than `python3Packages`, and exposing `buildPythonPackage` to the lexical scope right before using it is also unnecessary.

```suggestion
python3Packages.buildPythonPackage rec {
```